support/src/main/java/com/shopify/graphql/support/Input.java

```diff
@@ -0,0 +1,34 @@
+package com.shopify.graphql.support;
+
+import java.io.Serializable;
+
+/**
+ * Created by henrytao on 9/7/17.
+ */
+
+public final class Input<T> implements Serializable {
```

Comment:
Personally, I don't like final classes because it restricts what future developers can do and it makes mocking impossible

Reply:
Agree that in some cases it makes more sense to make them open, but not here as:

1. This is used internally in gen tool, what new features we are expecting to be added via inheritance and why?
2. This class is implemented in a way that explicitly restricts the inheritance as we it always adds complexity of correctly supporting inheritance. Plus  it's a value type (like sealed classes in Kotlin or enum in swift)
3. Mocking what you want to mock, 2 values? Why not just use instance without mocking.
4. From best practises by default classes should be closed (look for Kotlin, or Swift). And only make them open when you really need it and know what are you doing

Reply:
I guess it is mostly just a value class, I just don't see the value of restricting future developers from extending this. It's not like this is a super complicated class or is super critical.

I tend to default to trusting (future) developers to do the right thing.

Anyway, I am turning into a 🚲 🏠  discussion. This works fine